mesonbuild/msetup.py

```diff
@@ -180,6 +180,9 @@ def validate_dirs(self) -> T.Tuple[str, str]:
     # See class Backend's 'generate' for comments on capture args and returned dictionary.
     def generate(self, capture: bool = False, vslite_ctx: T.Optional[dict] = None) -> T.Optional[dict]:
         env = environment.Environment(self.source_dir, self.build_dir, self.options)
+        if not env.first_invocation:
+            assert self.options.reconfigure
+            env.coredata.set_from_configure_command(self.options)
```

Comment:
I think this technically means we "repeatedly" set this when using genvslite, but on the other hand it should be idempotent so do we actually care? There is currently no way to tell what the *first* run-through of that is.

Reply:
It's a problem if it triggers the assertion... I will double check what genvslite does and possibly remove it.

Reply:
genvslite uses multiple build directories and therefore multiple instances of `Environment`. Otherwise it would trigger the "Directory already configured." message.